### PR TITLE
test: dvm2 monitor

### PR DESCRIPTION
**Pull request by Reinis-FRP**
**Motivation**

Ensure DVM2.0 monitor bot is working as expected.

**Summary**

Adds tests to monitor bot modules for DVM2.0 monitor.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [x]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/monitor-v2/test/utils.ts [hunk 1]
```diff
@@ -10,6 +10,13 @@ export async function getContractFactory(contractName: ContractName, signer?: Si
   return new ContractFactory(contractAbi, contractBytecode, signer);
 }
 
-export const { formatBytes32String, parseUnits, toUtf8Bytes, toUtf8String } = utils;
+// Get block number from transaction (or 0 if transaction is not mined).
+export const getBlockNumberFromTx = async (tx: ContractTransaction): Promise<number> => {
+  await tx.wait();
+  const blockNumber = tx.blockNumber ? tx.blockNumber : 0;
```
Comment (mrice32) [inline packages/monitor-v2/test/utils.ts:16]: It doesn't really matter here, but in general, prefer !== undefined or !== null when dealing with an optional number to distinguish 0 from non existent/present value.

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/monitor-v2/test/utils.ts (hunk 1 above)
Comment (Reinis-FRP) [reply packages/monitor-v2/test/utils.ts]: Thanks, fixed in 2e98e4d32533e4c8405d29216fba529bb5049ae4

File: packages/monitor-v2/test/DVM2Monitor.ts [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+    const mintBlockNumber = await getBlockNumberFromTx(mintTx);
+
+    // Call monitorMints directly for the block when the mint was made.
+    const spy = sinon.spy();
+    const spyLogger = createNewLogger([new SpyTransport({}, { spy: spy })]);
+    await monitorMints(spyLogger, await createMonitoringParams(mintBlockNumber));
+
+    // When calling monitoring module directly there should be only one log (index 0) with the mint caught by spy.
+    assert.equal(spy.getCall(0).lastArg.at, "DVMMonitor");
+    assert.equal(spy.getCall(0).lastArg.message, "Large UMA minting 💸");
```
Comment (chrismaree) [inline packages/monitor-v2/test/DVM2Monitor.ts:419]: these are great.

File: packages/monitor-v2/test/OptimisticOracleV3Monitor.ts [hunk 3]
```diff
@@ -81,7 +83,7 @@ describe("OptimisticOracleV3Monitor", function () {
     const assertionTx = await optimisticOracleV3
       .connect(asserter)
       .assertTruthWithDefaults(claim, await asserter.getAddress());
-    const assertionBlockNumber = await getBlockNumber(assertionTx);
+    const assertionBlockNumber = await getBlockNumberFromTx(assertionTx);
```
Comment (chrismaree) [inline packages/monitor-v2/test/OptimisticOracleV3Monitor.ts:86]: 👍

**Review by chrismaree — APPROVED** [review]

**Pull request merged**